docs/notebooks/asynchronous_bayesian_optimization.pct.py

```diff
@@ -0,0 +1,373 @@
+                flush=True,
+            )
+            time.sleep(delay)
+
+        return scale * ((x1 - b * x0 ** 2 + c * x0 - r) ** 2 + s * (1 - t) * np.cos(x0) + translate)
+
+    observations = []
+    for point in points:
+        observation = scaled_branin(point)
+        observations.append(observation)
+
+    return np.array(observations)
+
+# test the defined objective function
+objective(np.array([[0.1, 0.5]]), sleep=False)
+
+# %% [markdown]
+# As always, we need to prepare model and some initial data to kick-start the optimization process.
+
+# %%
+from trieste.space import Box
+from trieste.data import Dataset
+from trieste.objectives import SCALED_BRANIN_MINIMUM
+
+search_space = Box([0, 0], [1, 1])
+num_initial_points = 3
+initial_query_points = search_space.sample(num_initial_points)
+initial_observations = objective(initial_query_points.numpy(), sleep=False)
+initial_data = Dataset(
+    query_points=initial_query_points,
+    observations=tf.constant(initial_observations, dtype=tf.float64),
+)
+
+import gpflow
+from trieste.models import create_model
+from trieste.models.gpflow.config import GPflowModelConfig
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.RBF(variance=variance)
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GPflowModelConfig(
+        **{
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+# these imports will be used later for optimization
+from trieste.acquisition import LocalPenalizationAcquisitionFunction
+from trieste.acquisition.rule import AsyncEfficientGlobalOptimization, EfficientGlobalOptimization
+from trieste.ask_tell_optimization import AskTellOptimizer
+
+
+# %% [markdown]
+# ## Multiprocessing setup
+#
+# To keep this notebook as reproducible as possible, we will only be using Python's multiprocessing package here. In this section we will explain our setup and define some common code to be used later.
+#
+# In both synchronous and asynchronous scenarios we will have a fixed set of worker processes performing observations. We will also have a main process responsible for optimization process with Trieste. When Trieste suggests a new point, it is inserted into a points queue. One of the workers picks this point from the queue, performs the observation, and inserts the output into observations queue. The main process then picks up the observation from the queue, at which moment it either waits for the rest of the points in the batch to come back (synchronous scenario) or immediately suggests a new point (asynchrunous scenario). This process continues either for a certain number of iterations or until we accumulate necessary number of observations.
```

Comment:
```suggestion
# In both synchronous and asynchronous scenarios we will have a fixed set of worker processes performing observations. We will also have a main process responsible for optimization process with Trieste. When Trieste suggests a new point, it is inserted into a points queue. One of the workers picks this point from the queue, performs the observation, and inserts the output into the observations queue. The main process then picks up the observation from the queue, at which moment it either waits for the rest of the points in the batch to come back (synchronous scenario) or immediately suggests a new point (asynchronous scenario). This process continues either for a certain number of iterations or until we accumulate necessary number of observations.
```

Reply:
"asynchronous" is tricky to spell out, thanks!